homeassistant/components/sensor/hmz19.py

```diff
@@ -0,0 +1,79 @@
+"""
+Support for CO2 sensor connected to a serial port.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.serial_pm/
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+
+REQUIREMENTS = ['pmsensor==0.3']
+
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_SERIAL_DEVICE = "serial_device"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=""): cv.string,
+    vol.Required(CONF_SERIAL_DEVICE): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the available CO2 sensors."""
```

Comment:
You should check if the device is working properly during setup and make the setup fail if the device is not present, working, or there is a communication error due to missing permissions. Otherwise the user will end up with a broken platform in their setup.